pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -413,7 +396,7 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return nil
 	}
 
-	for _, rprt := range pipelineState {
+	for _, rprt := range pipelineState.State() {
 		err := taskrun.ValidateResolvedTaskResources(rprt.PipelineTask.Params, rprt.ResolvedTaskResources)
```

Comment:
This check `ValidateResolvedTaskResources` can be moved to `ResolvePipelineRun()`.

Reply:
Yeah, it sounds like something we good do. Perhaps in different PR?

Reply:
yup definitely, not everything has to be part of single PR